### ISSUE TITLE
[Bug]: results of some queries of tpch 100G are not correct on eks with 3*cn

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#10799]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): f571de3
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2023-07-21 03:44:43 The query_id is: e8ea5a61-2778-11ee-987e-0647e3a1ea3f
**2023-07-21 03:44:43 Now start to compare query result for [report/TPCH_100/q4.txt,golden/TPCH_100/q4.txt]**
2023-07-21 03:44:43 The column[1] value of row[1] does not equal to each other, [1051859] and [1051977]
2023-07-21 03:44:43 The column[1] value of row[2] does not equal to each other, [1052864] and [1052987]
2023-07-21 03:44:43 The column[1] value of row[3] does not equal to each other, [1052808] and [1052921]
2023-07-21 03:44:43 The column[1] value of row[4] does not equal to each other, [1051368] and [1051493]
2023-07-21 03:44:43 The column[1] value of row[5] does not equal to each other, [1052079] and [1052194]
2023-07-21 03:44:43 The query q4  has failed to  be executed,and cost: 8706

2023-07-21 03:44:12 The query_id is: cbd5adc6-2778-11ee-987e-0647e3a1ea3f
**2023-07-21 03:44:12 Now start to compare query result for [report/TPCH_100/q21.txt,golden/TPCH_100/q21.txt]**
2023-07-21 03:44:12 The column[0] value of row[34] does not equal to each other, [Supplier#000798105] and [Supplier#000877025]
2023-07-21 03:44:12 The column[0] value of row[35] does not equal to each other, [Supplier#000877025] and [Supplier#000905481]
2023-07-21 03:44:12 The column[0] value of row[36] does not equal to each other, [Supplier#000905481] and [Supplier#000910925]
2023-07-21 03:44:12 The column[0] value of row[37] does not equal to each other, [Supplier#000910925] and [Supplier#000925795]
2023-07-21 03:44:12 The column[0] value of row[38] does not equal to each other, [Supplier#000925795] and [Supplier#000932203]
2023-07-21 03:44:12 The column[0] value of row[39] does not equal to each other, [Supplier#000932203] and [Supplier#000974242]
2023-07-21 03:44:12 The column[0] value of row[40] does not equal to each other, [Supplier#000974242] and [Supplier#000978306]
2023-07-21 03:44:12 The column[0] value of row[41] does not equal to each other, [Supplier#000978306] and [Supplier#000036540]
2023-07-21 03:44:12 The column[1] value of row[41] does not equal to each other, [21] and [20]
2023-07-21 03:44:12 The column[0] value of row[42] does not equal to each other, [Supplier#000036540] and [Supplier#000036927]
2023-07-21 03:44:12 The column[0] value of row[43] does not equal to each other, [Supplier#000036927] and [Supplier#000074440]
2023-07-21 03:44:12 The column[0] value of row[44] does not equal to each other, [Supplier#000074440] and [Supplier#000075576]
2023-07-21 03:44:12 The column[0] value of row[45] does not equal to each other, [Supplier#000075576] and [Supplier#000081093]
2023-07-21 03:44:12 The column[0] value of row[46] does not equal to each other, [Supplier#000081093] and [Supplier#000085005]
2023-07-21 03:44:12 The column[0] value of row[47] does not equal to each other, [Supplier#000085005] and [Supplier#000090521]
2023-07-21 03:44:12 The column[0] value of row[48] does not equal to each other, [Supplier#000090521] and [Supplier#000091064]
2023-07-21 03:44:12 The column[0] value of row[49] does not equal to each other, [Supplier#000091064] and [Supplier#000134375]
2023-07-21 03:44:12 The column[0] value of row[50] does not equal to each other, [Supplier#000134375] and [Supplier#000135014]
2023-07-21 03:44:12 The column[0] value of row[51] does not equal to each other, [Supplier#000135014] and [Supplier#000186358]
2023-07-21 03:44:12 The column[0] value of row[52] does not equal to each other, [Supplier#000186358] and [Supplier#000208724]
2023-07-21 03:44:12 The column[0] value of row[53] does not equal to each other, [Supplier#000208724] and [Supplier#000215094]
2023-07-21 03:44:12 The column[0] value of row[54] does not equal to each other, [Supplier#000215094] and [Supplier#000216727]
2023-07-21 03:44:12 The column[0] value of row[55] does not equal to each other, [Supplier#000216727] and [Supplier#000221620]
2023-07-21 03:44:12 The column[0] value of row[56] does not equal to each other, [Supplier#000221620] and [Supplier#000229300]
2023-07-21 03:44:12 The column[0] value of row[57] does not equal to each other, [Supplier#000229300] and [Supplier#000230486]
2023-07-21 03:44:12 The column[0] value of row[58] does not equal to each other, [Supplier#000230486] and [Supplier#000231704]
2023-07-21 03:44:12 The column[0] value of row[59] does not equal to each other, [Supplier#000231704] and [Supplier#000242129]
2023-07-21 03:44:12 The column[0] value of row[60] does not equal to each other, [Supplier#000242129] and [Supplier#000242937]
2023-07-21 03:44:12 The column[0] value of row[61] does not equal to each other, [Supplier#000242937] and [Supplier#000273371]
2023-07-21 03:44:12 The column[0] value of row[62] does not equal to each other, [Supplier#000273371] and [Supplier#000279173]
2023-07-21 03:44:12 The column[0] value of row[63] does not equal to each other, [Supplier#000279173] and [Supplier#000279912]
2023-07-21 03:44:12 The column[0] value of row[64] does not equal to each other, [Supplier#000279912] and [Supplier#000283812]
2023-07-21 03:44:12 The column[0] value of row[65] does not equal to each other, [Supplier#000283812] and [Supplier#000286340]
2023-07-21 03:44:12 The column[0] value of row[66] does not equal to each other, [Supplier#000286340] and [Supplier#000288362]
2023-07-21 03:44:12 The column[0] value of row[67] does not equal to each other, [Supplier#000288362] and [Supplier#000289179]
2023-07-21 03:44:12 The column[0] value of row[68] does not equal to each other, [Supplier#000289179] and [Supplier#000291168]
2023-07-21 03:44:12 The column[0] value of row[69] does not equal to each other, [Supplier#000291168] and [Supplier#000292152]
2023-07-21 03:44:12 The column[0] value of row[70] does not equal to each other, [Supplier#000292152] and [Supplier#000297866]
2023-07-21 03:44:12 The column[0] value of row[71] does not equal to each other, [Supplier#000297866] and [Supplier#000322460]
2023-07-21 03:44:12 The column[0] value of row[72] does not equal to each other, [Supplier#000322460] and [Supplier#000330583]
2023-07-21 03:44:12 The column[0] value of row[73] does not equal to each other, [Supplier#000330583] and [Supplier#000346138]
2023-07-21 03:44:12 The column[0] value of row[74] does not equal to each other, [Supplier#000346138] and [Supplier#000355805]
2023-07-21 03:44:12 The column[0] value of row[75] does not equal to each other, [Supplier#000355805] and [Supplier#000369722]
2023-07-21 03:44:12 The column[0] value of row[76] does not equal to each other, [Supplier#000369722] and [Supplier#000389504]
2023-07-21 03:44:12 The column[0] value of row[77] does not equal to each other, [Supplier#000389504] and [Supplier#000402088]
2023-07-21 03:44:12 The column[0] value of row[78] does not equal to each other, [Supplier#000402088] and [Supplier#000406722]
2023-07-21 03:44:12 The column[0] value of row[79] does not equal to each other, [Supplier#000406722] and [Supplier#000413613]
2023-07-21 03:44:12 The column[0] value of row[80] does not equal to each other, [Supplier#000413613] and [Supplier#000421782]
2023-07-21 03:44:12 The column[0] value of row[81] does not equal to each other, [Supplier#000421782] and [Supplier#000435940]
2023-07-21 03:44:12 The column[0] value of row[82] does not equal to each other, [Supplier#000435940] and [Supplier#000444902]
2023-07-21 03:44:12 The column[0] value of row[83] does not equal to each other, [Supplier#000444902] and [Supplier#000532668]
2023-07-21 03:44:12 The column[0] value of row[84] does not equal to each other, [Supplier#000532668] and [Supplier#000552183]
2023-07-21 03:44:12 The column[0] value of row[85] does not equal to each other, [Supplier#000552183] and [Supplier#000596205]
2023-07-21 03:44:12 The column[0] value of row[86] does not equal to each other, [Supplier#000596205] and [Supplier#000633454]
2023-07-21 03:44:12 The column[0] value of row[87] does not equal to each other, [Supplier#000633454] and [Supplier#000661988]
2023-07-21 03:44:12 The column[0] value of row[88] does not equal to each other, [Supplier#000661988] and [Supplier#000720862]
2023-07-21 03:44:12 The column[0] value of row[89] does not equal to each other, [Supplier#000720862] and [Supplier#000732109]
2023-07-21 03:44:12 The column[0] value of row[90] does not equal to each other, [Supplier#000732109] and [Supplier#000733274]
2023-07-21 03:44:12 The column[0] value of row[91] does not equal to each other, [Supplier#000733274] and [Supplier#000738376]
2023-07-21 03:44:12 The column[0] value of row[92] does not equal to each other, [Supplier#000738376] and [Supplier#0007[564](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/5613262440/job/15223495603#step:17:565)15]
2023-07-21 03:44:12 The column[0] value of row[93] does not equal to each other, [Supplier#000756415] and [Supplier#000774426]
2023-07-21 03:44:12 The column[0] value of row[94] does not equal to each other, [Supplier#000774426] and [Supplier#000785288]
2023-07-21 03:44:12 The column[0] value of row[95] does not equal to each other, [Supplier#000785288] and [Supplier#000798105]
2023-07-21 03:44:12 The query q21  has failed to  be executed,and cost: 2[590](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/5613262440/job/15223495603#step:17:591)0

2023-07-21 03:41:53 The query_id is: 84d1e600-2778-11ee-92b0-526c19fd9481
**2023-07-21 03:41:53 Now start to compare query result for [report/TPCH_100/q5.txt,golden/TPCH_100/q5.txt]**
2023-07-21 03:41:53 The column[1] value of row[1] does not equal to each other, [5311640295.7142] and [5311940984.3735]
2023-07-21 03:41:53 The column[1] value of row[2] does not equal to each other, [5279858396.9388] and [5280077133.5226]
2023-07-21 03:41:53 The column[1] value of row[3] does not equal to each other, [5278236070.2177] and [5278644512.1587]
2023-07-21 03:41:53 The column[0] value of row[4] does not equal to each other, [BRAZIL] and [PERU]
2023-07-21 03:41:53 The column[1] value of row[4] does not equal to each other, [5260356304.6277] and [5260560217.6528]
2023-07-21 03:41:53 The column[0] value of row[5] does not equal to each other, [PERU] and [BRAZIL]
2023-07-21 03:41:53 The column[1] value of row[5] does not equal to each other, [5260277430.3399] and [5260372003.3637]
2023-07-21 03:41:53 The query q5  has failed to  be executed,and cost: 5747

2023-07-21 03:35:24 The query_id is: 909218ce-2777-11ee-9ee6-daae9b2edd96
**2023-07-21 03:35:24 Now start to compare query result for [report/TPCH_100/q21.txt,golden/TPCH_100/q21.txt]**
2023-07-21 03:35:24 The column[0] value of row[15] does not equal to each other, [Supplier#000748793] and [Supplier#000896599]
2023-07-21 03:35:24 The column[0] value of row[16] does not equal to each other, [Supplier#000896599] and [Supplier#000083914]
2023-07-21 03:35:24 The column[1] value of row[16] does not equal to each other, [22] and [21]
2023-07-21 03:35:24 The column[0] value of row[17] does not equal to each other, [Supplier#000083914] and [Supplier#000111112]
2023-07-21 03:35:24 The column[0] value of row[18] does not equal to each other, [Supplier#000111112] and [Supplier#000142593]
2023-07-21 03:35:24 The column[0] value of row[19] does not equal to each other, [Supplier#000142593] and [Supplier#000225561]
2023-07-21 03:35:24 The column[0] value of row[20] does not equal to each other, [Supplier#000225561] and [Supplier#000238625]
2023-07-21 03:35:24 The column[0] value of row[21] does not equal to each other, [Supplier#000238625] and [Supplier#000302017]
2023-07-21 03:35:24 The column[0] value of row[22] does not equal to each other, [Supplier#000302017] and [Supplier#000310148]
2023-07-21 03:35:24 The column[0] value of row[23] does not equal to each other, [Supplier#000310148] and [Supplier#000371965]
2023-07-21 03:35:24 The column[0] value of row[24] does not equal to each other, [Supplier#000371965] and [Supplier#000372494]
2023-07-21 03:35:24 The column[0] value of row[25] does not equal to each other, [Supplier#000372494] and [Supplier#000400093]
2023-07-21 03:35:24 The column[0] value of row[26] does not equal to each other, [Supplier#000400093] and [Supplier#000535951]
2023-07-21 03:35:24 The column[0] value of row[27] does not equal to each other, [Supplier#000535951] and [Supplier#000571821]
2023-07-21 03:35:24 The column[0] value of row[28] does not equal to each other, [Supplier#000571821] and [Supplier#000576021]
2023-07-21 03:35:24 The column[0] value of row[29] does not equal to each other, [Supplier#000576021] and [Supplier#000641[145](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/5613262440/job/15223495603#step:17:146)]
2023-07-21 03:35:24 The column[0] value of row[30] does not equal to each other, [Supplier#000641145] and [Supplier#000684069]
2023-07-21 03:35:24 The column[0] value of row[31] does not equal to each other, [Supplier#000684069] and [Supplier#000748793]
2023-07-21 03:35:24 The query q21  has failed to  be executed,and cost: 26698

the jod link:
https://github.com/matrixorigin/mo-nightly-regression/actions/runs/5613262440/job/15223495603

the mo log can be get from loki by namespace:  mo-nightly-regression-20230721

**Comment by badboynt1** [comment]
2023-07-21 07:37:10 The query_id is: 63f95d47-2799-11ee-8e61-52972dda1fa1
2023-07-21 07:37:10 Now start to compare query result for [report/TPCH_100/q12.txt,golden/TPCH_100/q12.txt]
2023-07-21 07:37:10 The column[1] value of row[1] does not equal to each other, [623827] and [623835]
2023-07-21 07:37:10 The column[2] value of row[1] does not equal to each other, [936538] and [936551]
2023-07-21 07:37:10 The column[1] value of row[2] does not equal to each other, [624945] and [624953]
2023-07-21 07:37:10 The column[2] value of row[2] does not equal to each other, [937726] and [937736]
2023-07-21 07:37:10 The query q12  has failed to  be executed,and cost: 4592
补充一个eks上跑出来的错误

**Comment by badboynt1** [comment]
在eks上分析了一下，使用简化后的语句，只要一个join就可以复现出错误。 正确的行数是1560386， 有时候会错误输出1560355行。 

mysql> select         count(o_orderpriority)  from         orders,         lineitem where         o_orderkey = l_orderkey         and l_shipmode = 'FOB'
       and l_commitdate < l_receiptdate         and l_shipdate < l_commitdate         and l_receiptdate >= date '1996-01-01'         and l_receiptdate < date '1996-01-01' + interval '1' year;
+------------------------+
| count(o_orderpriority) |
+------------------------+
|                1560386 |
+------------------------+
1 row in set (4.19 sec)

mysql> select         count(o_orderpriority)  from         orders,         lineitem where         o_orderkey = l_orderkey         and l_shipmode = 'FOB'
       and l_commitdate < l_receiptdate         and l_shipdate < l_commitdate         and l_receiptdate >= date '1996-01-01'         and l_receiptdate < date '1996-01-01' + interval '1' year;
+------------------------+
| count(o_orderpriority) |
+------------------------+
|                1560355 |
+------------------------+
1 row in set (4.27 sec)

mysql> select         count(o_orderpriority)  from         orders,         lineitem where         o_orderkey = l_orderkey         and l_shipmode = 'FOB'
       and l_commitdate < l_receiptdate         and l_shipdate < l_commitdate         and l_receiptdate >= date '1996-01-01'         and l_receiptdate < date '1996-01-01' + interval '1' year;
+------------------------+
| count(o_orderpriority) |
+------------------------+
|                1560386 |
+------------------------+
1 row in set (4.20 sec)

**Comment by aunjgr** [comment]
怀疑是#10669 引入。正在eks上测试确认

**Comment by badboynt1** [comment]
eks环境暂时跑不了， 通过在129环境上模拟多cn的方式运行，已经验证确实是10669引入的问题

**Comment by aunjgr** [comment]
@aressu1985  PR已经合入，自测没有问题了。请确认

**Comment by aressu1985** [comment]
fixed:cbda0de71723cc4566e6136c4e67086004b509ea
